app/views/users/profile.json.jbuilder

```diff
@@ -11,7 +11,9 @@ else
     json.partial! 'performances/performance', performance: @next_performance
   end
 
-  if !@user.challenges.last.nil? && @user.challenges.last.performance.id == @next_performance.id
+  if @user.challenges.length <= 0 || @user.challenges.last.spot == @user.spot
```

Comment:
You should swap out `length` for `count` here so it uses a `COUNT` query instead of letting ruby check the length of the challenges.

Reply:
It's going to make the query later anyway, right? IMO, it's not that big of a deal

Reply:
Nope. It's grabbing the last one later. This query makes active record load all of the challenges into memory.

Reply:
All < 10 of them? 😝

Reply:
Unless you're passing all of the challenges somewhere else in this file (which I didn't look for)

Reply:
> Unless you're passing all of the challenges somewhere else in this file (which I didn't look for)

Yeah. Later on, all the challenges get rendered anyway